File: 1. Two Sum.js
//1. Two Sum


/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let sol = new Array();
    for(let i=0;i<nums.length;i++){
        for(let j=i+1;j<nums.length;j++){
            if(nums[i]+nums[j]==target){
                sol.push(i);
                sol.push(j);
                return sol;
            }
        }
    }
    return sol;
};

File: 1. Two Sum.java
//1. Two Sum



class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] sol = new int[2];
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int diff = target-nums[i];
            if(map.containsKey(diff)){
                sol[0]=i;
                sol[1]=map.get(diff);
                break;
            }
            map.put(nums[i],i);
        }
        return sol;
    }
}
